samples/snippets/src/main/java/com/example/dialogflow/ConversationProfileManagement.java

```diff
@@ -0,0 +1,252 @@
+    String conversationProfileId = 
+        ConversationProfileName.parse(conversationProfile.getName()).getConversationProfile();
+    getConversationProfile(projectId, location, conversationProfileId);
+
+    // Delete the conversation profile
+    deleteConversationProfile(projectId, location, conversationProfileId);
+  }
+
+  // [START dialogflow_list_conversation_profiles]
+  public static List<ConversationProfile> listConversationProfiles(
+      String projectId, String location) throws ApiException, IOException {
+    List<ConversationProfile> conversationProfiles = Lists.newArrayList();
+    try (ConversationProfilesClient conversationProfilesClient = 
+        ConversationProfilesClient.create()) {
+      LocationName locationName = LocationName.of(projectId, location);
+      for (ConversationProfile conversationProfile : 
+          conversationProfilesClient.listConversationProfiles(locationName).iterateAll()) {
+        System.out.println("====================");
+        System.out.format("Display name: %s\n", conversationProfile.getDisplayName());
+        System.out.format("Name: %s\n", conversationProfile.getName());
+        conversationProfiles.add(conversationProfile);
+      }
+    }
+    return conversationProfiles;
+  }
+  // [END dialogflow_list_conversation_profiles]
+  
+  // [START dialogflow_create_conversation_profile_article_faq]
+  public static ConversationProfile createConversationProfileArticleFaq(
+      String projectId, 
+      String displayName, 
+      String location,
+      Optional<String> articleSuggestionKnowledgeBaseId, 
+      Optional<String> faqKnowledgeBaseId) throws ApiException, IOException {
+    try (ConversationProfilesClient conversationProfilesClient = 
+        ConversationProfilesClient.create()) {
+      SuggestionConfig.Builder suggestionConfigBuilder = SuggestionConfig.newBuilder();
+
+      // Add knowledge base for article suggestion
+      if (articleSuggestionKnowledgeBaseId.isPresent()) {
+        KnowledgeBaseName articleSuggestionKbName = 
+            KnowledgeBaseName.of(projectId, articleSuggestionKnowledgeBaseId.get());
+        SuggestionFeatureConfig articleSuggestionFeatureConfig = 
+            SuggestionFeatureConfig.newBuilder()
+              .setSuggestionFeature(
+                SuggestionFeature.newBuilder().setType(Type.ARTICLE_SUGGESTION).build())
+              .setSuggestionTriggerSettings(
+                SuggestionTriggerSettings.newBuilder()
+                  .setNoSmalltalk(true).setOnlyEndUser(true).build())
+              .setQueryConfig(
+                SuggestionQueryConfig.newBuilder()
+                  .setKnowledgeBaseQuerySource(
+                    KnowledgeBaseQuerySource.newBuilder()
+                    .addKnowledgeBases(articleSuggestionKbName.toString()))
+                  .setMaxResults(3).build())
+              .build();
+        suggestionConfigBuilder.addFeatureConfigs(articleSuggestionFeatureConfig);
+      }
```

Comment:
These should _probably_ be separate similar snippets.

Reply:
I see. In this case maybe it's redundant to have list/get/delete examples for each resource, I will keep create method for now and update the structure. Would it make sense?